barretenberg/cpp/src/barretenberg/stdlib/primitives/biggroup/biggroup_nafs.hpp

```diff
@@ -11,6 +11,191 @@
 
 namespace bb::stdlib::element_default {
 
+template <typename C, class Fq, class Fr, class G>
+template <size_t wnaf_size>
+std::pair<uint64_t, bool> element<C, Fq, Fr, G>::compute_secp256k1_staggered_wnaf_fragment(const uint64_t fragment_u64,
+                                                                                           const uint64_t stagger,
+                                                                                           bool is_negative,
+                                                                                           bool wnaf_skew)
+{
+    // If there is not stagger then there is no need to change anyhing
+    if (stagger == 0) {
+        return std::make_pair<uint64_t, bool>((uint64_t)0, (bool)wnaf_skew);
+    }
+    int fragment = static_cast<int>(fragment_u64);
+    // Inverse the fragment if it's negative
+    if (is_negative) {
+        fragment = -fragment;
+    }
+    // If the value is positive and there is a skew in wnaf, subtract 2ˢᵗᵃᵍᵍᵉʳ. If negative and there is
```

Comment:
haha maybe this should just be:
```suggestion
    // If the value is positive and there is a skew in wnaf, subtract 2^{stagger}. If negative and there is
```